### PR TITLE
Refactor ferrocene bootstrap

### PR DESCRIPTION
**Pull request by Urhengulas**
Ferrocene uses the Rust bootstrap tool to build, test, sign, basically do everything with our tools. Before this PR there was one big file per subcommand. This PR splits up those big files into multiple submodules related to one component.

Blocked on https://github.com/ferrocene/ferrocene/pull/431.

### CODE REVIEW

File: .reuse/dep5 [hunk 1]
```diff
@@ -107,6 +107,10 @@ Copyright: 2014 Jorge Aparicio
            The Rust Project Developers (see https://thanks.rust-lang.org)
 License: MIT OR Apache-2.0
 
+Files: ferrocene/tools/flip-link/*
+Copyright: The Knurling-rs developers
```
[the remark this answers is not in the document]
Comment (pietroalbini) [reply .reuse/dep5:111]: For now I'd keep this as-is.

[the remark this answers is not in the document]
Comment (Urhengulas) [reply .reuse/dep5]: Opened a ticket for it: https://ferroussystems.clickup.com/t/86947yw8z

**Comment by Hoverbear** [comment]
By sheer volume and changes alone I worry about this PR's reviewability as well as losing changes while resolving merge conflicts. Perhaps you could choose to take each independent module split commit and we can quickly slip them through?

(This PR has also sat idle for 2 weeks, please either split it up, finish it & mark it for ready for review, or close it)

**Comment by Urhengulas** [comment]
> By sheer volume and changes alone I worry about this PR's reviewability as well as losing changes while resolving merge conflicts. Perhaps you could choose to take each independent module split commit and we can quickly slip them through?
> 
> (This PR has also sat idle for 2 weeks, please either split it up, finish it & mark it for ready for review, or close it)

Splitting it up is a good idea. Will close and open smaller ones.

**Pull request closed without merging**